U-boot patch: Copy RPI fw modified fdt settings to u-boot loaded fdt

leonardoaraujodf opened the pull request:
The closed source RPI firmware modifies several entries in the
device-tree that are important for proper boot. The complete list
of these entries can be found at:

https://forums.raspberrypi.com/viewtopic.php?p=1974233&hilit=firmware+device+tree+dma+ranges#p1974233

The function ft_board_setup() was modified so that if u-boot has
loaded a fdt that is different than the rpi fw fdt, then
it will copy all those important nodes and properties
created and modified by the rpi fw to the u-boot loaded fdt
blob.

Signed-off-by: Leonardo Amorim <lam@vizrt.com>

Commit:
U-boot patch: Copy RPI fw modified fdt settings to u-boot loaded fdt

The closed source RPI firmware modifies several entries in the
device-tree that are important for proper boot. The complete list
of these entries can be found at:

https://forums.raspberrypi.com/viewtopic.php?p=1974233&hilit=firmware+device+tree+dma+ranges#p1974233

The function ft_board_setup() was modified so that if u-boot has
loaded a fdt that is different than the rpi fw fdt, then
it will copy all those important nodes and properties
created and modified by the rpi fw to the u-boot loaded fdt
blob.

Signed-off-by: Leonardo Amorim <lam@vizrt.com>

cdsteinkuehler commented on patches/uboot/0002-arm-rpi-copy-rpi-fw-modified-fdt-settings-to-u-boot-.patch:
```diff
@@ -0,0 +1,331 @@
++/*
++ * Here's a list of node manipulations that the Rpi firmware does to proper boot
++ * the Linux Kernel.
++ * If not using the option `fw_dtb` in Kernel bootargs, the dtb passed to the
```
fw_dtb only exists in the context of the br2rauc u-boot script so should probably not be mentioned here.  Instead, add a more generic comment about the Linux kernel needing the device-tree modifications made by the closed-source RPi firmware.

leonardoaraujodf replied:
Removed comment mentioning the `fw_dtb`. Added a more generic comment about the u-boot loaded device-tree being missing some important entries to boot the Linux Kernel when it's not using the Rpi Firmware loaded device-tree.

cdsteinkuehler commented on patches/uboot/0002-arm-rpi-copy-rpi-fw-modified-fdt-settings-to-u-boot-.patch:
```diff
@@ -0,0 +1,331 @@
+ };
+ 
++/*
++ * Here's a list of node manipulations that the Rpi firmware does to proper boot
```
s/proper/properly/

leonardoaraujodf replied:
Thank you! Solved!

cdsteinkuehler commented on patches/uboot/0002-arm-rpi-copy-rpi-fw-modified-fdt-settings-to-u-boot-.patch:
```diff
@@ -0,0 +1,331 @@
++		while (nodeoffset == -FDT_ERR_NOTFOUND) {
++			/* The fw fdt node was not found in u-boot fdt and it
++			 * must be created.
++			 * First found the fw node parent offset.
```
s/found/find/

leonardoaraujodf replied:
Solved! I have made the same mistake at other comments.

cdsteinkuehler reviewed (changes requested):
Looks great!
Just two minor issues with English (your English is generally very good!) and the fw_dtb comment really doesn't belong here.

While you're fixing those, I'll try to test a patched U-Boot here and make sure everything looks like it's working as expected.

leonardoaraujodf commented:
Hi! Thank you for the notes! I have just submitted a amend for this commit and I changed the comments to correct the English mistakes. In my RPI4 seems to be working fine.

Commit:
Remove commands to copy specific device-tree entries

The .ush scripts were using specific commands to copy
entries from the Rpi firmware loaded fdt to the u-boot
loaded fdt. Remove those because now the u-boot handle
those copies internally in the function ft_board_setup().

Signed-off-by: Leonardo Amorim <lam@vizrt.com>

Commit:
Edit u-boot patch 0002 to copy only specific entries

An internal structure has been created to specify which
properties to copy from each node. Previously when a
node was specified, all properties from that node
were copied, which is not required.

Signed-off-by: Leonardo Amorim <lam@vizrt.com>
